### PR TITLE
G Suite: Remove Upsell from signup flow

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request seeks to remove the upsell page for G Suite from the signup flow. Note this page is still displayed when an existing user purchases a new domain for their site:

screenshot

#### Testing instructions

1. Run `git checkout remove/gsuite-upsell-signup` and start your server, or open a [live branch](https://calypso.live/?branch=remove/gsuite-upsell-signup)
2. Open the [`Signup` page](http://calypso.localhost:3000/start)
3. Select a domain and a plan, and purchase them with credits
4. Assert that you are not presented with the upsell page for G Suite
5. Navigate to the [`Domain Search` page](http://calypso.localhost:3000/domains/add)
6. Select a domain
7. Assert that you are presented with the upsell page for G Suite

#### Notes

@travisw, @niranjan-uma-shankar: removing this nudge will logically increase the visibility of other nudges, such as the one for the Concierge. Please let me know if that's a problem.

### CODE REVIEW

File: client/my-sites/checkout/checkout/composite-checkout-thank-you.ts [hunk 1]
```diff
@@ -208,81 +194,8 @@ function getFallbackDestination( {
...
-}
-
-function getRedirectUrlForPlanBumpOffer( { pendingOrReceiptId, cart, siteSlug } ) {
-	if ( hasPersonalPlan( cart ) ) {
-		if ( 'variantShowPlanBumpOffer' === abtest( 'showPlanBumpVsGsuite' ) ) {
-			return `/checkout/${ siteSlug }/offer-plan-upgrade/premium/${ pendingOrReceiptId }`;
-		}
-	}
-	return null;
-}
```
Comment (stephanethomas) [inline client/my-sites/checkout/checkout/composite-checkout-thank-you.ts]: @victorespigares: I think this method and some of the code that relates to https://github.com/Automattic/wp-calypso/pull/38680 should have been removed in https://github.com/Automattic/wp-calypso/pull/39296. Note I'm not sure how changes to the existing and new checkouts are supposed to be handled/synced, maybe @payton or @nbloomf could clarify that.

File: client/my-sites/checkout/checkout/composite-checkout-thank-you.ts [hunk 2]
```diff
@@ -128,17 +124,6 @@ export function getThankYouPageUrl( {
 		return `${ newBlogUrl }/${ pendingOrReceiptId }`;
 	}
 
-	const redirectPathForGSuiteUpsell = getRedirectUrlForGSuiteNudge( {
-		pendingOrReceiptId,
-		siteSlug,
-		cart,
-		didPurchaseFail,
-		isNewlyCreatedSite,
```
Comment (stephanethomas) [inline client/my-sites/checkout/checkout/composite-checkout-thank-you.ts]: @sirbrillig, @nbloomf : `didPurchaseFail` and `isNewlyCreatedSite` are no longer used in this component, should we remove them?

**Review by klimeryk — APPROVED** [review]
Tested and LGTM

**Review by niranjan-uma-shankar — APPROVED** [review]
LGTM.

Minor thing, but we can also remove the following lines:
1. [gsuite display mode param](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/index.jsx#L619)
2. GSuite messages in the [checklist](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checklist/main.jsx#L81) and [customer home](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/customer-home/main.jsx#L166).

Feel free to merge without review after making these changes.

There's a lint error but I see that you are [clarifying on that message](https://github.com/Automattic/wp-calypso/pull/39744#discussion_r385271005).

**Comment by niranjan-uma-shankar** [comment]
> removing this nudge will logically increase the visibility of other nudges, such as the one for the Concierge. Please let me know if that's a problem.

I recommend giving a heads up to HEs in the `business-concierge` slack channel. The increased visibility might not be a problem, since HEs usually have a couple of ways to handle high volumes. Knowing about this change could help them handle things better.

**Comment by stephanethomas** [comment]
@klimeryk, @niranjan-uma-shankar: thank your for your review! I've addressed feedback and also fixed linting and E2E tests.

**Pull request merged**